keras/saving/save_test.py

```diff
@@ -385,9 +385,20 @@ def _assert_same_weights_and_metrics(self, model, loaded_model):
       if testing_utils.get_save_format() == 'tf':
         # TODO(b/153110928): Keras TF format doesn't restore optimizer weights
         # currently.
-        return
-      self.assertAllClose(model.optimizer.weights,
-                          loaded_model.optimizer.weights)
+        try:
+          self.assertAllClose(model.optimizer.weights,
+                            loaded_model.optimizer.weights)
+        except (AssertionError, ValueError):
+          # Both errors can be raised, depending on the if shapes happen to be the same or not
+          return  # expected failure, but the test can't continue
+        else:
+          raise AssertionError(
+            "This test should have failed since SavedModel does not support restoring"
+            " optimizer weights"
+          )
+      else:
+        self.assertAllClose(model.optimizer.weights,
```

Comment:
Honestly I would move this in a separate test function decorating it with https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/eager/def_function_test.py#L243 with a reference to the Github ticket. 

So when it is implemented the test will fail and we have a pointer to the original issue/feature request.
Then we could remove the decorator or simplify/unify the original test (the current method).

Reply:
@adriangb Can you handle this comment?

Reply:
I don't think I'll have bandwidth anytime soon

Reply:
No problem the comment itself is 6 months old